"make generic" broken

f2a53fa3cbae8341f02a2f76caf209ed1bf453be broke `make -f Makefile.legacy generic`. This fixes it:

```diff
+++ b/src/bench.c
@@ -382,6 +382,7 @@ char *benchmark_format(struct fmt_main *format, int salts,
 #ifdef BENCH_BUILD
        const char *where;
        if ((where = fmt_self_test(format, test_db))) {
+               static char s_error[128];
                snprintf(s_error, sizeof(s_error), "FAILED (%s)\n", where);
                return s_error;
        }
```

Found while trying to build for an arch not yet recognized by our `configure`... to fix which we need to make our `configure.ac` friendly to newer autoconf.

There's also:

```
bench.c: In function 'benchmark_all':
bench.c:804: warning: unused variable 'using_int_mask'
bench.c:680: warning: unused variable 'i'
```

So maybe:

```diff
+++ b/src/bench.c
@@ -382,6 +382,7 @@ char *benchmark_format(struct fmt_main *format, int salts,
 #ifdef BENCH_BUILD
        const char *where;
        if ((where = fmt_self_test(format, test_db))) {
+               static char s_error[128];
                snprintf(s_error, sizeof(s_error), "FAILED (%s)\n", where);
                return s_error;
        }
@@ -676,7 +677,6 @@ void gather_results(struct bench_results *results)
 int benchmark_all(void)
 {
        struct fmt_main *format;
-       int i;
 #if defined(HAVE_OPENCL)
        char s_gpu[16 * MAX_GPU_DEVICES] = "";
        char s_gpu1[16 * MAX_GPU_DEVICES] = "";
@@ -685,6 +685,7 @@ int benchmark_all(void)
        const char *s_gpu1 = "";
 #endif
 #ifndef BENCH_BUILD
+       int i;
        unsigned int loop_fail = 0, loop_total = 1;
        int nvidia_mem = 0;
 #endif
@@ -799,7 +800,7 @@ AGAIN:
 #if defined(HAVE_OPENCL) || defined(HAVE_ZTEX)
                int using_int_mask = (format->params.flags & FMT_MASK) && (options.flags & FLG_MASK_CHK) &&
                        options.req_int_cand_target != 0 && mask_int_cand_target;
-#else
+#elif !defined(BENCH_BUILD)
                int using_int_mask = 0;
 #endif
 
```

And there's also this:

```
wordlist.c: In function 'do_wordlist_crack':
wordlist.c:576: error: 'S_IFMT' undeclared (first use in this function)
wordlist.c:576: error: (Each undeclared identifier is reported only once
wordlist.c:576: error: for each function it appears in.)
wordlist.c:576: error: 'S_IFIFO' undeclared (first use in this function)
```

@AlekseyCherepanov You want to fix this?

--- a/src/bench.c
+++ b/src/bench.c
@@ -382,6 +382,7 @@ char *benchmark_format(struct fmt_main *format, int salts,
 #ifdef BENCH_BUILD
 	const char *where;
 	if ((where = fmt_self_test(format, test_db))) {
+		static char s_error[128];
 		snprintf(s_error, sizeof(s_error), "FAILED (%s)\n", where);
 		return s_error;
 	}
@@ -676,7 +677,6 @@ void gather_results(struct bench_results *results)
 int benchmark_all(void)
 {
 	struct fmt_main *format;
-	int i;
 #if defined(HAVE_OPENCL)
 	char s_gpu[16 * MAX_GPU_DEVICES] = "";
 	char s_gpu1[16 * MAX_GPU_DEVICES] = "";
@@ -685,6 +685,7 @@ int benchmark_all(void)
 	const char *s_gpu1 = "";
 #endif
 #ifndef BENCH_BUILD
+	int i;
 	unsigned int loop_fail = 0, loop_total = 1;
 	int nvidia_mem = 0;
 #endif
@@ -799,7 +800,7 @@ AGAIN:
 #if defined(HAVE_OPENCL) || defined(HAVE_ZTEX)
 		int using_int_mask = (format->params.flags & FMT_MASK) && (options.flags & FLG_MASK_CHK) &&
 			options.req_int_cand_target != 0 && mask_int_cand_target;
-#else
+#elif !defined(BENCH_BUILD)
 		int using_int_mask = 0;
 #endif
 

--- a/src/loader.c
+++ b/src/loader.c
@@ -1933,7 +1933,9 @@ static void ldr_fill_user_words(struct db_main *db)
 	} else {
 		map_pos = mem_map;
 		map_end = mem_map + file_len;
+#if MGETL_HAS_SIMD
 		map_scan_end = map_end - VSCANSZ;
+#endif
 	}
 #endif /* HAVE_MMAP */
 

--- a/src/mgetl.h
+++ b/src/mgetl.h
@@ -14,11 +14,12 @@
 #if defined(vcmpeq_epi8_mask) && !defined(_MSC_VER) && !VLOADU_EMULATED
 #define MGETL_HAS_SIMD          1
 #define VSCANSZ                 sizeof(vtype)
+static char *map_scan_end;
 #else
 #define VSCANSZ                 0
 #endif
 
-static char *mem_map, *map_pos, *map_end, *map_scan_end;
+static char *mem_map, *map_pos, *map_end;
 
 #define GET_LINE(line, file)	(mem_map ? mgetl(line) : fgetl(line, LINE_BUFFER_SIZE, file))
 

--- a/src/wordlist.c
+++ b/src/wordlist.c
@@ -13,14 +13,14 @@
 #if AC_BUILT
 #include "autoconfig.h"
 #else
-#ifndef sparc
-#undef _POSIX_SOURCE
-#define _POSIX_SOURCE /* for fileno(3) */
+#ifndef _XOPEN_SOURCE
+#define _XOPEN_SOURCE /* for fileno(3) and stat(2) */
 #endif
 #endif
 
 #include <stdint.h>
 #include <stdio.h>
+#include <sys/types.h>
 #include <sys/stat.h>
 
 #if (!AC_BUILT || HAVE_UNISTD_H) && !_MSC_VER
@@ -638,7 +638,9 @@ void do_wordlist_crack(struct db_main *db, const char *name, int rules)
 			} else {
 				map_pos = mem_map;
 				map_end = mem_map + file_len;
+#if MGETL_HAS_SIMD
 				map_scan_end = map_end - VSCANSZ;
+#endif
 			}
 		}
 #endif

